app/controllers/forest_liana/actions_controller.rb

```diff
@@ -46,7 +46,14 @@ def handle_result(result, formatted_fields, action)
       end
 
       # Apply result on fields (transform the object back to an array), preserve order.
-      fields = action.fields.map { |field| result[field[:field]] }
+      fields = action.fields.map do |field|
+        f = result[field[:field]]
+        # Reset `value` when not present in `enums` (which means `enums` has changed).
+        if f[:enums].is_a?(Array) && !f[:enums].include?(f[:value])
+          f[:value] = nil
+        end
+        f
```

Comment:
Not sure it's better but we tend to use full names (ignorable):

```suggestion
        updated_field = result[field[:field]]
        # Reset `value` when not present in `enums` (which means `enums` has changed).
        if updated_field[:enums].is_a?(Array) && !updated_field[:enums].include?(updated_field[:value])
          updated_field[:value] = nil
        end
        updated_field
```